src/axom/sidre/core/MFEMSidreDataCollection.hpp

```diff
@@ -299,6 +304,57 @@ class MFEMSidreDataCollection : public mfem::DataCollection
       deleting it from the named_buffers group, if allocated. */
   virtual void DeregisterField(const std::string& field_name);
 
+  /// Associates a field name with a multi-buffer material set
+  /** Subsequent calls to RegisterField with field_names of the form
+   * @p volume_fraction_field_name_<material_id> will result in the addition
+   * of a volume buffer to the matset @p matset_name corresponding to @p material_id.
+   * 
+   * Note that this does not inhibit the addition of the field - that is, the GridFunction
+   * data will be present as both a field and a volume fraction buffer
+   * 
+   * @param volume_fraction_field_name The field name to associate with volume fractions
+   * @param matset_name The material set to associate subsequently added volume fractions with
+   *
+   */
+  void AssociateMaterialSet(const std::string& volume_fraction_field_name,
+                            const std::string& matset_name);
+
+  /// Associates a field name with a species set
+  /** Subsequent calls to RegisterField with field_names of the form
+   * @p species_field_name_<material_id>_<component> will result in the addition
+   * of its values to the specset @p specset_name corresponding to @p material_id.
+   * and specified component
+   * 
+   * Note that this does not inhibit the addition of the field - that is, the GridFunction
+   * data will be present as both a field and within the species set
+   * 
+   * @param species_field_name The field name to associate with the species set matset values
+   * @param specset_name The name of the species set to associate added matset values with
+   * @param matset_name The material set to associate with the species set
+   * @param volume_dependent Whether the species set is volume-dependent
+   *
```

Comment:
Is it an error to call this if  `matset_name` was not registered first via `AssociateMaterialSet()` ?

Reply:
I turned this into a warning, see https://github.com/LLNL/axom/pull/473#discussion_r625823847